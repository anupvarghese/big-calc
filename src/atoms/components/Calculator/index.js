import React, { Component } from 'react';
import { css } from 'glamor';
import Keypad from '../Keypad';
import Display from '../Display';

class Calculator extends Component {
  constructor() {
    super();
    this.state = {
      display: 0,
    };
  }

  handleKeyClick = tag => () => {
    this.setState({
      clickedTag: tag,
      display: Number(this.state.display + tag),
    });
  };

  render() {
    return (
      <div
        {...css({
          display: 'flex',
          flexDirection: 'column',
          width: '300px',
        })}
      >
        <Display value={this.state.display} />
        <Keypad handleKeyClick={this.handleKeyClick} />
      </div>
    );
  }
}

export default Calculator;
