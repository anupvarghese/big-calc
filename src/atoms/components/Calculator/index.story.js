import React from 'react';

import { storiesOf } from '@storybook/react';

import Calculator from './';

storiesOf('Calculator').add('show Calculator', () => <Calculator />);
