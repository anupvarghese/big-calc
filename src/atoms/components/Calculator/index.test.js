import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-glamor-react';

import Calculator from './';

expect.addSnapshotSerializer(serializer);

describe('Calculator', () => {
  it('renders Calculator', () => {
    const calculator = renderer.create(<Calculator />).toJSON();
    expect(calculator).toMatchSnapshot();
  });
});
