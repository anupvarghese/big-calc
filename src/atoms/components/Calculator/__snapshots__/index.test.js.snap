// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders Calculator 1`] = `
.css-1rh9gak,
[data-css-1rh9gak] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  width: 300px;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-9v06sd,
[data-css-9v06sd] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
}

.css-rvzscm,
[data-css-rvzscm] {
  min-height: 4rem;
  flex-basis: 100%;
  border-radius: 4px;
  border: 1px solid tomato;
  font-size: 1.5rem;
  -webkit-flex-basis: 100%;
}

.css-rvzscm:focus,
[data-css-rvzscm]:focus {
  outline: none !important;
  border: 1px solid blue;
  box-shadow: 0 0 10px #719ECE;
  border-radius: 4px;
}

.css-tcqo1y,
[data-css-tcqo1y] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  border: 1px solid tomato;
  border-radius: 4px;
  margin-top: 0.5rem;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-1qeez79,
[data-css-1qeez79] {
  flex: 2;
  -webkit-flex: 2;
}

.css-q8ffom,
[data-css-q8ffom] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  flex: 2;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-flex: 2;
}

.css-h0wfqv,
[data-css-h0wfqv] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  text-align: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-8g7ilp,
[data-css-8g7ilp] {
  flex: 1;
  margin: 0.5rem;
  border: 1px solid tomato;
  padding: 1rem;
  pointer-events: ;
  -webkit-flex: 1;
}

<div
  data-css-1rh9gak=""
>
  <div
    data-css-9v06sd=""
  >
    <input
      data-css-rvzscm=""
      disabled={true}
      type="text"
      value=""
    />
  </div>
  <div
    data-css-tcqo1y=""
  >
    <div
      data-css-h0wfqv=""
    >
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        AC
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        +/-
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        %
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        /
      </div>
    </div>
    <div
      data-css-h0wfqv=""
    >
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        7
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        8
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        9
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        X
      </div>
    </div>
    <div
      data-css-h0wfqv=""
    >
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        4
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        5
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        6
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        -
      </div>
    </div>
    <div
      data-css-h0wfqv=""
    >
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        1
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        2
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        3
      </div>
      <div
        data-css-8g7ilp=""
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <div
      data-css-h0wfqv=""
    >
      <div
        data-css-1qeez79=""
      >
        <div
          data-css-8g7ilp=""
          onClick={[Function]}
        >
          0
        </div>
      </div>
      <div
        data-css-q8ffom=""
      >
        <div
          data-css-8g7ilp=""
          onClick={[Function]}
        >
          .
        </div>
        <div
          data-css-8g7ilp=""
          onClick={[Function]}
        >
          =
        </div>
      </div>
    </div>
  </div>
</div>
`;
